refactor: reorganize files to follow popular convention

utsabc opened the pull request:
## Description of the change

Refactoring transformer repo

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

## Checklists

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

saikumarrs commented:
A few more things to fix/refactor:
- Move all the tests and related files to `test`.
- Rename `destTransformer.js` with `index.js`.
- Take out `benchmark` and `scripts` from the `src` directory.
- Review the `package.json` file for the impact due to directory changes (`test:benchmark` was missed).

utsabc commented:
> A few more things to fix/refactor:
> 
>     * Move all the tests and related files to `test`.
> 
>     * Rename `destTransformer.js` with `index.js`.
> 
>     * Take out `benchmark` and `scripts` from the `src` directory.
> 
>     * Review the `package.json` file for the impact due to directory changes (`test:benchmark` was missed).

@saikumarrs I have addressed these points

saikumarrs commented on CHANGELOG.md:
```diff
@@ -7,212 +7,190 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
```
Since this file is auto-generated, it should be excluded from the formatter and linters.

saikumarrs commented on CONTRIBUTING.md:
```diff
@@ -46,9 +46,9 @@ message.
+
 [slack]: https://resources.rudderstack.com/join-rudderstack-slack
 [issue]: https://github.com/rudderlabs/rudder-transformer/issues/new
-[CLA]: https://rudderlabs.wufoo.com/forms/rudderlabs-contributor-license-agreement
```
Restore this change and also exclude this file from the linter and formatter.

koladilip commented:
@utsabc please check the build image failure, otherwise fine
